modules/probe.sh

```diff
@@ -13,6 +13,29 @@
 #    See the License for the specific language governing permissions and
 #    limitations under the License.
 
+# Checks there is no preexisting mender integration on the golden image
+#
+# mender-convert expects a vanilla image wihtout mender installed by other means.
+# If you provide as input an image that has mender installed by other means already
+# it might fail.
+# This function will just print warnings for traces of things that look like a tainted input image
+probe_untainted_rootfs_input() {
+    # None of these should be present in a clean rootfs
+    directories=(
+        "work/rootfs/var/lib/mender-monitor"
+        "work/rootfs/var/lib/mender-configure"
+        "work/data/mender"
+    )
+
+    for dir in "${directories[@]}"; do
+        if [ -e "$dir" ]; then
+            log_warn "$dir exists!"
+            log_warn "Input image contain traces Mender of previous Mender installation"
+            log_warn "The conversion might fail. Please provide a clean input image."
+        fi
+    done
```

Comment:
I would suggest something like this instead to avoid hardcoding the paths:
```suggestion
    directories="$(sudo find work/ -type d -name "mender*")"
    ## None of these should be present in a clean rootfs
    if [ -n "$directories" ]; then
        log_warn "Found following directories:\n$directories"
        log_warn "Input image contains traces of Mender"
        log_warn "The conversion might fail. Please provide a clean input image."
    fi
```

Reply:
>  I'm worried of its potential to slow down the conversion

Good point. I don't know how much extra time it adds, but I can agree with you in that it might not be worth it just to print a warning. 

Squash the two commits and I'll approve this 👍